tests/optional_kernel_features/kernel_features_device_has_exceptions.cpp

```diff
@@ -0,0 +1,330 @@
+
+      run_functor<FunctorT>(is_exception_expected, expected_errc, queue);
+    }
+
+    {
+      RUN_SUBMISSION_CALL(is_exception_expected, expected_errc, queue,
+                          NO_ATTRIBUTE,
+                          use_feature_function_non_decorated<FeatureTypeT>());
+    }
+  }
+
+#ifdef SYCL_EXTERNAL
+  SECTION(
+      "Kernel does not have the attribute [[sycl::device_has()]] but it "
+      "calls a SYCL_EXTERNAL function which uses the tested feature. The "
+      "SYCL_EXTERNAL function is defined in another translation unit. The "
+      "SYCL_EXTERNAL function is declared with the corresponding attribute "
+      "[[sycl::device_has()]].") {
+    {
+      const auto lambda_no_arg = []() {
+        use_feature_function_external_decorated<FeatureTypeT, FeatureAspectT>();
+      };
+      const auto lambda_item_arg = [](sycl::item<1>) {
+        use_feature_function_external_decorated<FeatureTypeT, FeatureAspectT>();
+      };
+      const auto lambda_group_arg = [](sycl::group<1>) {
+        use_feature_function_external_decorated<FeatureTypeT, FeatureAspectT>();
+      };
+
+      run_separate_lambda(is_exception_expected, expected_errc, queue,
+                          lambda_no_arg, lambda_item_arg, lambda_group_arg);
+    }
+
+    {
+      using FunctorT =
+          non_decorated_call_decorated_external_function<FeatureTypeT,
+                                                         FeatureAspectT>;
+
+      run_functor<FunctorT>(is_exception_expected, expected_errc, queue);
+    }
+
+    {
+      RUN_SUBMISSION_CALL(
+          is_exception_expected, expected_errc, queue, NO_ATTRIBUTE,
+          use_feature_function_external_decorated<FeatureTypeT,
+                                                  FeatureAspectT>());
+    }
+  }
+#endif
+
+  SECTION(
+      "Kernel does not use the tested feature but is decorated with the "
+      "corresponding attribute [[sycl::device_has()]].") {
+    {
+      const auto lambda_no_arg = []() [[sycl::device_has(FeatureAspectT)]] {
+        dummy_function_non_decorated();
+      };
+      const auto lambda_item_arg =
+          [](sycl::item<1>) [[sycl::device_has(FeatureAspectT)]] {
+        dummy_function_non_decorated();
+      };
+      const auto lambda_group_arg =
+          [](sycl::group<1>) [[sycl::device_has(FeatureAspectT)]] {
+        dummy_function_non_decorated();
+      };
+
+      run_separate_lambda(is_exception_expected, expected_errc, queue,
+                          lambda_no_arg, lambda_item_arg, lambda_group_arg);
+    }
+
+    {
+      using FunctorT = decorated_call_non_decorated_dummy<FeatureAspectT>;
+
+      run_functor<FunctorT>(is_exception_expected, expected_errc, queue);
+    }
+
+    {
+      RUN_SUBMISSION_CALL(is_exception_expected, expected_errc, queue,
+                          [[sycl::device_has(FeatureAspectT)]],
+                          dummy_function_non_decorated());
+    }
+  }
+
+  SECTION(
```

Comment:
Does it make sense to move "sections" to separate "test cases"? Will it enable compilation of some test cases with any of implementations?

Reply:
I think it makes sense. 
This will allow us to enable most of the tests from this PR for DPCPP, but not for other implementations. 
ComputeCpp still has an issue with linkage (Philip mentioned that [here](https://github.com/KhronosGroup/SYCL-CTS/pull/285#issuecomment-1054035325)) and it does not support unnamed kernels yet.
hipSYCL does not have a definition of `sycl::errc`.
So it will not enable compilation of some tests with any implementation.

Anyway I spilt sections into different test cases in this commit:  https://github.com/KhronosGroup/SYCL-CTS/pull/287/commits/e297e1e06527e7bcf1a477eeb46e2deb5edaa383